x/mongo/driver/topology/topology_options.go

```diff
@@ -71,31 +71,76 @@ func newLogger(opts *options.LoggerOptions) (*logger.Logger, error) {
 	return log, nil
 }
 
-// NewConfig will translate data from client options into a topology config for building non-default deployments.
-func NewConfig(co *options.ClientOptions, clock *session.ClusterClock) (*Config, error) {
-	// Auth & Database & Password & Username
-	if co.Auth != nil {
-		cred := &auth.Cred{
-			Username:    co.Auth.Username,
-			Password:    co.Auth.Password,
-			PasswordSet: co.Auth.PasswordSet,
-			Props:       co.Auth.AuthMechanismProperties,
-			Source:      co.Auth.AuthSource,
+// convertOIDCArgs converts the internal *driver.OIDCArgs into the equivalent
+// public type *options.OIDCArgs.
+func convertOIDCArgs(args *driver.OIDCArgs) *options.OIDCArgs {
+	if args == nil {
+		return nil
+	}
+	return &options.OIDCArgs{
+		Version:      args.Version,
+		IDPInfo:      (*options.IDPInfo)(args.IDPInfo),
+		RefreshToken: args.RefreshToken,
+	}
+}
+
+// NewAuthenticator returns a [driver.Authenticator] configured with the given
+// credential and HTTP client. It returns nil if cred is nil.
+func NewAuthenticator(cred *options.Credential, httpClient *http.Client) (driver.Authenticator, error) {
```

Comment:
I suggest repurposing this function with the goal of converting `options.Credential` to `auth.Cred`. And constructing the `Authenticator` using the `auth` package where needed. For example, 

```go
authCreds := topology.ConvertCreds(clientOpt.Auth)

client.authenticator, err = auth.CreateAuthenticator(clientOpt.Auth.AuthMechanism, authCreds, clientOpt.HTTPClient)
```

This way we limit the number of functions that return the `Authenticator` interface, making a future refactor to correct this anti-pattern more straightforward: GODRIVER-3316

Reply:
Seems reasonable, will try that refactor.